src/ORM/Locator/TableLocator.php

```diff
@@ -299,4 +306,29 @@ public function remove($alias)
             $this->_fallbacked[$alias]
         );
     }
+
+    /**
+     * Returns a singleton instance of LocatorInterface implementation.
+     *
+     * @return \Cake\ORM\Locator\LocatorInterface
+     */
+    public static function getInstance()
```

Comment:
This PR aims to reduce the number of static calls in userland code, to discourage from using `TableRegistry` and move to the underlying implementation of table locators.

The static call is done only once when the `getTableLocator()` is called from an objecy using the trait.